kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/workitem/WorkItemDefinitionCacheRegistry.java

```diff
@@ -36,10 +36,7 @@ public WorkItemDefinitionCacheRegistry() {
 
     @Override
     public Collection<WorkItemDefinition> items() {
-        return definitions
-                .values()
-                .stream()
-                .collect(Collectors.toList());
+        return new ArrayList<>(definitions.values());
```

Comment:
Hey @hasys 
I'm not agree on this change for different reasons:
* You're forcing to use as an implementation the concrete ArrayList type, any reason for it? On the other hand by delegating this to the Collections API the data collection, it can decide for us what's the best type to create at this point (it depends on the runtime stuff, architecture, the payload size, etc)
* If at some point this code is going to be executed in parallel, you're forced again to change all these lines everywhere, because as you know an ArrayList is not thread-safe, so again, better letting the Collections API to collect and generate the result. 

In general the Sreams API is designed to support multi-threading out-of-the-box, if you ensure the immutability of the input arguments you use, all the functions are supposed to be pure, to you can safely turn-on multi-threading at any point and it ensures the code will run with no issues. Same for the collect method, it uses internal collections to populate the resulting collection but it does in a really safe manner, it ensures the objects it creates to handle the collection all are immutable as well. So IMO it's a much better way to handle the collections than using "old school" new operators... 

I do this explain as you're proposing this change each time I create a PR... but I do not agree, and I'm giving you my reasons why no to do that. So until you don't give me some reason about why better creating an ArrayList rather than using the Collections API to construct the resulting object, I'll not stop refusing this change.. :)